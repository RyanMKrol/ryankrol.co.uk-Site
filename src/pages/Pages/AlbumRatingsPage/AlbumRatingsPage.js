import React, { Component } from 'react';

import { DataCollection, AlbumRatingsItem } from '../../../components';

import styles from './AlbumRatingsPage.module.css';

import { baseEndpoint, generateCustomSorter } from '../../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class AlbumRatingsPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.endpoint = `${baseEndpoint()}/ratings/album`;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h2>Albums I've Listened To</h2>
        <br />
        <DataCollection
          endpoint={this.endpoint}
          itemTag={AlbumRatingsItem}
          sorter={generateCustomSorter('rating')}
        />
      </div>
    );
  }
}

export default AlbumRatingsPage;
