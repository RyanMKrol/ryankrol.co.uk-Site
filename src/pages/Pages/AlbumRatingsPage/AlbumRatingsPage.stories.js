import React from 'react';

import AlbumRatingsPage from './AlbumRatingsPage';

export default {
  title: 'Pages/AlbumRatingsPage',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the AlbumRatingsPage
 */
export const normal = () => <AlbumRatingsPage />;
