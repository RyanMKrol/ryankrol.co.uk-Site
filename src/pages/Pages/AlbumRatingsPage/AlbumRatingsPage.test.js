import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import AlbumRatingsPage from './AlbumRatingsPage';

it('renders AlbumRatingsPage correctly', () => {
  const header = shallow(<AlbumRatingsPage />);
  expect(toJson(header)).toMatchSnapshot();
});
