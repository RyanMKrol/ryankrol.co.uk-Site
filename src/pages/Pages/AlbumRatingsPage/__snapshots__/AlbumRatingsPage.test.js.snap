// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AlbumRatingsPage correctly 1`] = `
<div
  className="container"
>
  <h2>
    Albums I've listened to
  </h2>
  <br />
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/ratings/album"
    itemTag={[Function]}
    sorter={[Function]}
  />
</div>
`;
